### DISCUSSION TITLE
Is it possible to use as a library?

### DISCUSSION THREAD

**Discussion by ghost** [lcn2/calc#110] [Q&A] [answered]
Like `libqalculate`?

https://github.com/Qalculate/libqalculate

**Comment by lcn2** [comment, marked as answer]
Yes, calc builds libraries from which you can link operations.

See the file `LIBRARY` for detailed information or run the calc command:

```calc
; help libcalc
```

In addition, nearly all help files for builtin functions contains a section called "LINK LIBRARY" that lists the libcalc functions that as needed to perform the function.  For example:

```calc
; help sin
```

Calc comes with 2 examples of programs that link to libcalc.  See the file `sample.README` and the source files `sample_many.c` and `sample_rand.c` for two examples of this.

**Comment by lcn2** [comment]
BTW: Calc does have a custom interface for those who wish to call their own code within the calc framework.

See the file `custom/CUSTOM_CAL` or run the calc command:

```calc
; new_custom
```

The entire `custom/` tree is devoted to examples of custom functions.  Moreover when you install calc, there is a separate library `libcustcalc` that contains those custom function examples and whatever other code you may have linked into that system.

Some people use the custom interface to access custom hardware accelerators, for example.  Other people write special calculation operations to improve the algorithmic performance or to make use of an external mathematical library within the calc framework.
